### PR TITLE
Import datasets and documentation from epidatasets

### PR DESCRIPTION
**Pull request by nmdefries**
### Checklist

Please:

- [x] Make sure this PR is against "dev", not "main" (unless this is a release
      PR).
- [x] Request a review from one of the current main reviewers:
      brookslogan, nmdefries.
- [x] Makes sure to bump the version number in `DESCRIPTION`. Always increment
      the patch version number (the third number), unless you are making a
      release PR from dev to main, in which case increment the minor version
      number (the second number).
- [x] Describe changes made in NEWS.md, making sure breaking changes
      (backwards-incompatible changes to the documented interface) are noted.
      Collect the changes under the next release number (e.g. if you are on
      1.7.2, then write your changes under the 1.8 heading).
- See [DEVELOPMENT.md](DEVELOPMENT.md) for more information on the development
  process.

### Change explanations for reviewer

Reexport datasets from `epidatasets`, inheriting all associated documentation. Add data attribution -- since license info is included in dataset documentation, this is just adding data contributors to DESCRIPTION.

Supersedes https://github.com/cmu-delphi/epiprocess/pull/344

Closes https://github.com/cmu-delphi/epiprocess/issues/327

### CODE REVIEW

**Comment by nmdefries** [comment]
Failing `document` is because the `epidatasets` version on `main` doesn't export all of the same data objects.

**Comment by nmdefries** [comment]
This also needs to rebuild docs and add `doctor-visits` attribution to the DESCRIPTION, once we have that from Roni.

File: vignettes/correlation.Rmd [hunk 1]
```diff
@@ -16,36 +16,48 @@ state-level COVID-19 case and death rates, smoothed using 7-day trailing
...
-library(epidatr)
 library(epiprocess)
 library(dplyr)
 ```
 
-The data is fetched with the following query:
-```{r, message = FALSE}
+The data is included in this package (via the [`epidatasets` package](https://cmu-delphi.github.io/epidatasets/)) and can be loaded with:
+```{r}
+x <- cases_deaths_subset %>%
```
Comment (brookslogan) [inline vignettes/correlation.Rmd:25]: question: This isn't the same data; this vignette apparently used all states, not just 6.  Was this change intentional?  (& if so, have you already checked sanity-checked new results and any statements about dimensions of the input?)

**Review by brookslogan — APPROVED** [review]
Pushed some tweaks; please
- [ ] double-check tweaks

Only blocker to merging:
- [x] double-checking whether correlation vignette changes were intentional/okay.  If they weren't intentional & you want me to check if changes are okay, please ping me / re-request review.

Others comments are ideas for the future; could you refile to issues if you think it makes sense?  Not sure what doc overhaul is going to look like.

**Comment by brookslogan** [comment]
Wait, there is also
- [ ] Update DESCRIPTION and NEWS.md given PRs that have been merged.  Version should be 0.9.{x+1} this will go under 0.10 changelog heading.

File: vignettes/correlation.Rmd (hunk 1 above)
Comment (nmdefries) [reply vignettes/correlation.Rmd:25]: Not intentional, thanks for pointing thisout

File: vignettes/correlation.Rmd [hunk 2]
```diff
@@ -16,36 +16,48 @@ state-level COVID-19 case and death rates, smoothed using 7-day trailing
...
+  select(geo_value, time_value, case_rate = case_rate_7d_av, death_rate = death_rate_7d_av) %>%
```
[the remark this answers is not in the document]
Comment (nmdefries) [reply vignettes/correlation.Rmd]: The correct example dataset already has the `case_rate`/`death_rate` names, so no need to rename anymore.

**Comment by nmdefries** [comment]
This now depends on https://github.com/cmu-delphi/epidatasets/pull/5

**Review by brookslogan — APPROVED** [review]
Added a bit more to the NEWS.md entry; merging.

**Comment by brookslogan** [comment]
Looks like CHECKs have some bugs and we have nolints covering some up.  Trying to fix up now.

**Comment by brookslogan** [comment]
We inherit some soft CHECK complaints from https://github.com/cmu-delphi/epidatasets/issues/10 but I'm going to go ahead and merge.
